### PR TITLE
feat: API stitch 

### PR DESCRIPTION
**Pull request by subhan97ahmed**
renamed action folder to utils

### CODE REVIEW

File: src/utils/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ */
+import { message } from "antd";
+import axios from "axios";
+
+// We need to set default config for axios
+axios.defaults.baseURL = "https://fake-news-detector-backend.herokuapp.com";
+axios.defaults.headers.post['Access-Control-Allow-Origin'] = '*'
+axios.defaults.headers.post['accept'] = "application/json"
+axios.defaults.headers.post["Content-Type"] =
+  "application/json";
```
Comment (MuhammadAnas190) [inline src/utils/index.js]:
```suggestion
axios.defaults.headers.post = { 
  'Access-Control-Allow-Origin': '*',
  'accept': 'application/json',
  'Content-Type': 'application/json',
}
```
We can move this into a single object rather having separate definition

File: src/utils/index.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+// We will be using async function for dispatching network requests
+export async function fetchNewsResult(newsText) {
+  try {
+    const response = await axios.post('/detect', {
+      text: newsText
+    });
+    console.log(response.data);
+    message.success('Calculated Successfully')
+    const result = await response.data
+    return result;
```
Comment (MuhammadAnas190) [inline src/utils/index.js]:
```suggestion
    return response.data;
```
We don't need await on `response.data`. This is not a promise.

File: src/view/Main.jsx [hunk 3]
```diff
@@ -11,7 +11,9 @@ const { Header, Content } = Layout;
 const { Title, Text } = Typography;
 
 /**
- * @author MuhammadAnas190
+ * @authors
+ * MuhammadAnas190
+ * Subhan Ahmed
```
Comment (MuhammadAnas190) [inline src/view/Main.jsx]: FYI, git knows which line you add and how much you have contributed, This comment was just for info. Best approach would be to remove the @authors rather mentioning your name

File: src/view/Main.jsx [hunk 4]
```diff
@@ -20,6 +22,38 @@ const { Title, Text } = Typography;
...
+    // temp = temp.replace("/[\n]/g", "");
+    // temp = temp.replace(/[+]/g, " ");
+    // temp = temp.replace(/[^]/g, "");
+    // temp = temp.replace(/[$]/g, "");
+    // temp =temp.replace(/[^\x20-\x7E]/gmi, "")
+    temp = temp.replace(/(\r\n|\n|\r)/gm, " ").trim();
+    temp = temp.replace(/[â€™]/g, "");
+    temp = temp.replace(/[â€œ]/g, "");
+    temp = temp.replace(/[]/g, "");
+    // temp = temp.replace(/[^a-zA-Z0-9]/g, '');
```
Comment (MuhammadAnas190) [inline src/view/Main.jsx]: Remove commented code and please let me know why using so many replace methods? Either create a single regex. that will remove all the special symbols. Also, don't lower the case of the news text.

File: src/view/Main.jsx [hunk 5]
```diff
@@ -20,6 +22,38 @@ const { Title, Text } = Typography;
...
+
+    console.log(temp);
+    setnewsText(temp);
+    if (newsText) {
+      message.loading("Calculating news");
+      const tempr = await fetchNewsResult(newsText);
+      setresult(tempr);
```
Comment (MuhammadAnas190) [inline src/view/Main.jsx]: Update const name to something which make more sense

File: src/view/Main.jsx [hunk 6]
```diff
@@ -43,6 +77,34 @@ const MainView = (props) => {
...
+                  <HXTextArea
+                    placeholder="News here"
+                    onChange={OnChangeHXTextArea}
+                    size="large"
+                    style={{ height: "150px" }}
+                  />
+                </Row>
+                <HXButton
+                  onClick={onClickSubmit}
+                  style={{ float: "right", marginTop: "10px" }}
```
Comment (MuhammadAnas190) [inline src/view/Main.jsx]: Don't add style attribute. CSS file is created already. Use `className` to add styles. Every where you have added `style` attribute.

File: src/view/Main.jsx [hunk 7]
```diff
@@ -43,6 +77,34 @@ const MainView = (props) => {
                   is the correct article. We are using Tensorflow model through
                   which we will be finding out the fake news.
                 </Text>
+                <Row>
+                  <HXTextArea
+                    placeholder="News here"
+                    onChange={OnChangeHXTextArea}
+                    size="large"
+                    style={{ height: "150px" }}
```
Comment (MuhammadAnas190) [inline src/view/Main.jsx:85]: create a css class

File: src/view/Main.jsx [hunk 8]
```diff
@@ -43,6 +77,34 @@ const MainView = (props) => {
...
+                <HXButton
+                  onClick={onClickSubmit}
+                  style={{ float: "right", marginTop: "10px" }}
+                >
+                  Analyze
+                </HXButton>
+                <Row>
+                  <Col>
+                    {result.prediction === "Real" ? (
+                      <p style={{ color: "green", marginTop: "10px" }}>
```
Comment (MuhammadAnas190) [inline src/view/Main.jsx:97]: create a css class

File: src/view/Main.jsx [hunk 9]
```diff
@@ -43,6 +77,34 @@ const MainView = (props) => {
...
+                        Its Real
+                      </p>
+                    ) : null}
+                    {result.prediction === "Fake" ? (
+                      <p style={{ color: "red", marginTop: "10px" }}>
```
Comment (MuhammadAnas190) [inline src/view/Main.jsx:102]: create a css class

File: src/view/Main.jsx [hunk 10]
```diff
@@ -43,6 +77,34 @@ const MainView = (props) => {
...
+                        Its Fake
+                      </p>
+                    ) : null}
```
Comment (MuhammadAnas190) [inline src/view/Main.jsx]:
```suggestion
                    {result.prediction === "Real" ? (
                      <Text 
                        className="result-true"
                        // style={{ color: "green", marginTop: "10px" }} Use this style in CSS
                      > 
                        Its Real
                      </Text>
                    ) : result.prediction === "Fake" ? (
                      <Text 
                        className="result-false"
                        // style={{ color: "red", marginTop: "10px" }}
                      >
                        Its Fake
                      </Text>
                    ) : <Text>Umm! Something went wrong...</Text>}
```

**Review by MuhammadAnas190 — CHANGES_REQUESTED** [review]
Please update the requested fixes

File: src/view/Main.jsx (hunk 7 above)
Comment (MuhammadAnas190) [inline src/view/Main.jsx]: Don't use style height. Check ant design docs and update it number of rows.

File: src/view/Main.jsx (hunk 4 above)
Comment (subhan97ahmed) [reply src/view/Main.jsx]: because there is a difference in the type of regex and news text needs to be in lowercase

File: src/view/Main.jsx [hunk 11]
```diff
@@ -10,16 +10,32 @@ import "view/main.css";
...
-/**
- * @author MuhammadAnas190
- * We will be setting a basic layout for the web.
- * Also we should be calling our actions from here
- * and pass down the values through props (single responsibility)
- *
- * @todo make sure to not use "style" attribute. Create classes which should be inherited from parent class
- * .main-view
- */
 const MainView = (props) => {
```
Comment (MuhammadAnas190) [inline src/view/Main.jsx:22]:
```suggestion
/**
 * We will be setting a basic layout for the web.
 * Also we should be calling our actions from here
 * and pass down the values through props (single responsibility)
 *
 * @todo make sure to not use "style" attribute. Create classes which should be inherited from parent class
 * .main-view
 */

const MainView = (props) => {
```

**Review by MuhammadAnas190 — APPROVED** [review]
Good work!

**Comment by subhan97ahmed** [comment]
> Good work!

thank you.

**Pull request merged**